src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -272,18 +269,16 @@ module T = struct
       ~constraint_constants ~pending_coinbase_collection =
     { ledger; scan_state; constraint_constants; pending_coinbase_collection }
 
-  let of_scan_state_and_ledger ~logger
```

Comment:
This is called during bootstrap. Shouldn't we be doing the statement check then? I did it in the snapps branch https://github.com/MinaProtocol/mina/blob/82c6b4eb87fb5c2e00e649b722709c2c955c878e/src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml#L435